src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportPump.cs

```diff
@@ -0,0 +1,230 @@
+        _transportBindingElement = transportBindingElement;
+        _securityBindingElement = securityBindingElement;
+        _baseAddress = serviceDispatcher.BaseAddress;
+    }
+
+    private ConsumerConfig BuildConfig()
+    {
+        var config = new ConsumerConfig
+        {
+            BootstrapServers = _baseAddress.Authority,
+            IsolationLevel = IsolationLevel.ReadCommitted,
+            AutoOffsetReset = AutoOffsetReset.Earliest
+        };
+
+        foreach (var keyValue in _transportBindingElement.Config)
+        {
+            config.Set(keyValue.Key, keyValue.Value);
+        }
+
+        foreach (var keyValue in _securityBindingElement.Config)
+        {
+            config.Set(keyValue.Key, keyValue.Value);
+        }
+
+
+        var (enableAutoCommit, enableAutoOffsetStore) = GetCommitStrategyConfigValues(config, _kafkaDeliverySemantics);
+        config.EnableAutoCommit = enableAutoCommit;
+        config.EnableAutoOffsetStore = enableAutoOffsetStore;
+
+        return config;
+    }
+
+    private static (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) GetCommitStrategyConfigValues(ConsumerConfig consumerConfig, KafkaDeliverySemantics kafkaDeliverySemantics) =>
+        (kafkaDeliverySemantics, consumerConfig) switch
+        {
+            // KafkaBinding
+            (KafkaDeliverySemantics.AtMostOnce, { EnableAutoCommit: null, EnableAutoOffsetStore: null }) => s_atMostOnceConfigValues,
+            (KafkaDeliverySemantics.AtLeastOnce, { EnableAutoCommit: null, EnableAutoOffsetStore: null } ) => s_atLeastOnceBatchCommitConfigValues,
+            // CustomBinding
+            (KafkaDeliverySemantics.AtMostOnce, { EnableAutoCommit: false, EnableAutoOffsetStore: null }) => s_atMostOnceConfigValues,
+            (KafkaDeliverySemantics.AtLeastOnce, { EnableAutoCommit: true, EnableAutoOffsetStore: false } ) => s_atLeastOnceBatchCommitConfigValues,
+            (KafkaDeliverySemantics.AtLeastOnce, { EnableAutoCommit: false, EnableAutoOffsetStore: null }) => s_atLeastOncePerMessageCommitConfigValues,
+            _ => throw new NotSupportedException(string.Format(SR.ProvidedKafkaConfigurationIsNotValid, kafkaDeliverySemantics, consumerConfig.EnableAutoCommit, consumerConfig.EnableAutoOffsetStore))
+        };
+
+    public override Task StartPumpAsync(QueueTransportContext queueTransportContext, CancellationToken token)
+    {
+        _cts = new();
+        _mres = new();
+        _mres.Reset();
+
+        CancellationToken ct = CancellationTokenSource
+            .CreateLinkedTokenSource(_cts.Token, token).Token;
+
+        ConsumerConfig config = BuildConfig();
+        _consumer = new ConsumerBuilder<Null, byte[]>(config)
+            .SetKeyDeserializer(Deserializers.Null)
+            .SetValueDeserializer(Deserializers.ByteArray)
+            .SetLogHandler(OnLog)
+            .SetErrorHandler(OnError)
+            .Build();
+
+        _consumer.Subscribe(_topic);
```

Comment:
@birojnayak Indeed this is a nice feature to consume message from multiple topics which each own a specific ser/des logic. In CoreWCF, i'm wondering if we need this because we are already able to "mix" different message types in the same topic. The Dispatch logic of CoreWCF is handling the routing of the message to the correct implementation. Is it something that has been implemented in MSMQ / RabbitMQ ?

Reply:
My preference would be make sure we don't block ourselves in with the design and add this if people ask for it. In other words, make sure we wouldn't require anyone to change an existing usage if it was added later.

Reply:
@g7ed6e the diff between RabbitMQ (or any Queue) vs notification(Kafka) is Kafka provides out of the box subscription to listen multiple topics while RabbitMQ client doesn't. I looked at your recent changes, I think below way we can extend the feature for multiple subscription.... We can expose another property with List<String> topics, and under startpump logic where background task is run can be spawned per topic. Or let me know what's your thought on extending ?  Bdw I am not asking to implement now (can be a future Feature/Back log issue), I am trying to answer @mconnew question about future extensibility without impacting previous version.

Also , can the background task to consume message in StartPumpAsync can be moved to another async method?

Reply:
@birojnayak  Confluent.Kafka .net allows to subscribe to multiples topics by passing a regex to the `Consumer.Subscribe()` method. It allows to have one thread processing the consumer loop (under the hood librdkafka is using multiple threads) and dispatch messages (multithreaded if needed).
https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.IConsumer-2.html#Confluent_Kafka_IConsumer_2_Subscribe_System_String_
Please notice:
- in latest changes I made I added a regex to ensure we fail fast if topic name has special character or length>255. 
- the topic name is passed within the address
```c#
services.AddServiceEndpoint<TestService, ITestContract>(new KafkaBinding
                {
                    AutoOffsetReset = AutoOffsetReset.Earliest,
                    DeliverySemantics = KafkaDeliverySemantics.AtMostOnce,
                    GroupId = consumerGroupAccessor.Invoke(),
                    Security = new KafkaSecurity
                    {
                        Mode = KafkaSecurityMode.Transport,
                        Transport = new KafkaTransportSecurity
                        {
                            CredentialType = KafkaCredentialType.None,
                            CaPem = SslHelper.GetSslCaPem()
                        }
                    }
                }, $"net.kafka://localhost:9093/my-topic");
```
I think we can later add an enum property to the binding to let user choose between either a plain text or a regex.